src/features/Chat/ChatMenu.tsx

```diff
@@ -0,0 +1,158 @@
+import { useTranslation } from 'react-i18next';
+
+import BackArrowIcon from '@/static/icons/back-arrow.svg';
+
+import IconButton from '@/components/Buttons/IconButton';
+import { palette } from '@/components/variables';
+import Text from '@/components/Text';
+
+type ChatCategory = 'active' | 'archived' | 'blocked';
+
+const ChatMenu = () => {
+  const { t } = useTranslation('chat');
+  const [showCategories, setShowCategories] = useState(false);
+  const [currentCategory, setCurrentCategory] =
+    useState<ChatCategory>('active');
+  const [chats, setChats] = useState([]);
+
+  return (
+    <Container>
+      <Row>
+        <Header variant="h1">
+          {currentCategory === 'active' && t('menu.title.chats')}
+          {currentCategory === 'archived' && t('menu.title.archived')}
+          {currentCategory === 'blocked' && t('menu.title.blocked')}
+        </Header>
+        <Buttons>
+          {!!chats.length && (
+            <IconButton
+              variant="search"
+              sizeInPx={40}
+              onClick={() => console.log('searching...')}
+            />
+          )}
+          <IconButton
+            variant="menuLines"
+            sizeInPx={40}
+            onClick={() => setShowCategories(!showCategories)}
+          />
+        </Buttons>
+      </Row>
+
+      {(showCategories ||
+        currentCategory === 'archived' ||
+        currentCategory === 'blocked') && (
+        <Row>
+          <GoBackLink
+            onClick={() => {
+              setCurrentCategory('active');
+              setShowCategories(false);
+            }}
+          >
+            <GoBackIcon src={BackArrowIcon} />
+            <Text color="purple" variant="bold">
+              {t('menu.back')}
+            </Text>
+          </GoBackLink>
+        </Row>
+      )}
+
+      {showCategories ? (
+        <>
+          <Row>
+            <CategoryLink
+              onClick={() => {
+                setCurrentCategory('archived');
+                setShowCategories(false);
+              }}
+            >
+              <Text color="purple" variant="bold">
+                {t('menu.archived')}
+              </Text>
+            </CategoryLink>
+          </Row>
+          <Row>
+            <CategoryLink
+              onClick={() => {
+                setCurrentCategory('blocked');
+                setShowCategories(false);
+              }}
+            >
+              <Text color="purple" variant="bold">
+                {t('menu.blocked')}
+              </Text>
+            </CategoryLink>
+          </Row>
+        </>
+      ) : chats.length ? (
+        <ChatList></ChatList>
+      ) : (
+        <CategoryEmptyText variant="p">
+          {currentCategory === 'active' && t('menu.empty.chats')}
+          {currentCategory === 'archived' && t('menu.empty.archived')}
+          {currentCategory === 'blocked' && t('menu.empty.blocked')}
+        </CategoryEmptyText>
+      )}
+    </Container>
+  );
+};
+
```

Comment:
This return is way too big imo. We should abstract this to multiple components.

Reply:
It would be _ideal_ if you could see the logic and the returned components in a small glance without having to scroll a lot

Reply:
Yeah true! The menu and all its functionality is not ready yet so this too will be split in the upcoming PRs. 👍